File: hacker_rank/week_3/betweenTwoSets.js

// a = multiples of a
// b = divisible by b

// find the largest number within the arrays to have a max stopping point for all loops
// find the similar multiple of a[i] up to largest
// fing the similar multiples of b[i] from the largest to 0


const getTotal = (a, b) => {
    let result = 0;
    let isFactorMultiple = true;
    const length = a.length >= b.length ? a.length : b.length;
    let maxA = a[0];
    let minB = b[0];


    for (let i = 0; i < length; i++) {
        if (a[i] > maxA) {
            maxA = a[i];
        }
        if (b[i] < minB) {
            minB = b[i];
        }
    }

    for (let j = maxA; j <= minB; j++) {
        for (let elem of a) {
            if (j % elem !== 0) {
                isFactorMultiple = false;
                // break
            }
        }
        for (let elem of b) {
            if (elem % j !== 0) {
                isFactorMultiple = false;
                // break
            }
        }
        if (isFactorMultiple === true) {
            result++
        }
    }

    return result

}


console.log(getTotal([2, 6], [24, 36]));
console.log(getTotal([2, 3], [2, 4]));

File: hacker_rank/week_3/betweenTwoSets.py


def getTotalX(a, b):
    result = 0
    
    print(max(a))
    print(min(b))
    for i in range(max(a), min(b)+1):
        isFactorMultiple = True
        
        for ele in a:
            print(a)
            if i % ele != 0:
                isFactorMultiple = False
                break
            
        for ele in b:
            if ele % i != 0:
                isFactorMultiple = False
                break
                
        if isFactorMultiple == True:
            result += 1
                
                
    return result



getTotalX([2, 6], [24, 36]);
getTotalX([2, 3], [2, 4]);